cala-ledger/src/entry/mod.rs

```diff
@@ -96,6 +96,47 @@ impl Entries {
         Ok(entries)
     }
 
+    pub async fn new_entries_for_voided_tx(
+        &self,
+        original_tx_id: TransactionId,
+        new_tx_id: TransactionId,
+    ) -> Result<Vec<NewEntry>, EntryError> {
+        let entries = self.list_for_transaction_id(original_tx_id).await?;
+
+        let new_entries = entries
+            .into_iter()
+            .map(|entry| {
+                let value = entry.into_values();
+                let mut units = value.units;
+                units.set_sign_negative(true);
```

Comment:
I think this is wrong - we need to flip the sign not set it negative. (It might already be negative).